src/common/shared_lib.cc

```diff
@@ -0,0 +1,74 @@
+#include "shared_lib.h"
+
+
+#if defined(_WIN32)
```

Comment:
@petrutlucian94 instead of introducing a set of functions for manipulating shared libraries, have you considered adding an other approach? something like: 

1. add a `dlfcn_win32.h` and `dlfcn_win32.cc`  for Win32. 
1. add a library named `win32::dl` for Win32, so and make sure instead of `-ldl` on Win32 this library is linked.

Reply:
Yes, but we can't have a drop in replacement because the Windows API is slightly different (e.g. error message handling or dlopen flags) so it's probably better to have an abstraction on top rather than a lib trying to copy the semantics.

Reply:
@petrutlucian94 i assume your concern is `shared_lib_last_err()` and `shared_lib_free_err_msg()` pair. if that's the case we could have something like:
```c++
#ifdef WIN32
using last_error_t = std::string;
HMODULE dlopen(const char* filename, int /* flags */) {
  return  LoadLibrary(filename);
}

last_error_t dlerror() {
  DWORD err_code = ::GetLastError();
  LPSTR buffer = NULL;
  DWORD len = FormatMessageA(...);
  assert(len > 0);
  std::string s{buffer, len);
  LocalFree(buffer);
  return s;
}
#endif
```

regarding to "dlopen flags", IIUC, you just drop the flags in Windows implementation.

Reply:
You're right, that should work. I'll give it a try and come with a follow up.

Reply:
I sent a follow up. Thanks a lot for this suggestion, it's much cleaner now.

True that dlerror now returns std::string instead of char* but that's a drop in replacement as well in this case.